lib/apiservers/engine/backends/container.go

```diff
@@ -476,6 +473,36 @@ func (c *ContainerBackend) ContainerExecStart(ctx context.Context, eid string, s
 		op.Errorf("Failed to start Exec task for container(%s) due to error (%s)", id, err)
 		return err
 	}
+
+	// now that the exec has begun and the streams are attached we need to inspect and then wait.
+	op.Debugf("Waiting for completion: task(%s), container(%s)", eid, name)
+	handle, err := c.Handle(id, name)
+	if err != nil {
+		op.Errorf("Failed to obtain handle during exec start for container(%s) due to error: %s", id, err)
+		return engerr.InternalServerError(err.Error())
+	}
+
+	ec, err := c.containerProxy.InspectTask(op, handle, eid, id)
```

Comment:
Check whether any of the retry errors can appear here. Ideally the core transient ones (host disconnected, et al) should be retried inside this call but needs confirming.